components/usergroup/readMore.tsx

```diff
@@ -1,31 +1,60 @@
 import classNames from "classnames";
-import { useEffect, useState } from "react";
+import { useEffect, useMemo, useState } from "react";
 import { MdOutlineKeyboardArrowDown } from "react-icons/md";
 import { sanitiseXSS } from "../../helpers/validator";
 
 type ReadMoreProps = {
   text: string;
-  length: number;
+  length?: number;
+  previewSentenceCount?: number;
   className?: string;
 };
 
-const condenseContent = (text: string, length: number) => {
-  const formatted = text.split(". ")[0] + ". ...";
+const getCondensedText = (
+  text: string,
+  length: number,
+  previewSentenceCount: number
+) => {
+  const formatted = text.split(". ");
+  const selectedParagraphs = formatted.slice(0, previewSentenceCount);
+  const condensedText = selectedParagraphs.join(". ");
 
-  if (formatted.length <= length) return formatted;
-  else return formatted.slice(0, length) + "...";
+  if (length > 0 && condensedText.length <= length) {
+    return `${condensedText} . ...`;
+  } else if (length > 0) {
+    return `${condensedText.slice(0, length)}...`;
+  }
+
+  return `${condensedText}. ...`;
 };
 
-export const ReadMore = ({ text, length, className }: ReadMoreProps) => {
+export const ReadMore = ({
+  text,
+  length,
+  className,
+  previewSentenceCount = 1,
+}: ReadMoreProps) => {
   const [dropdownClicked, setDropdownClicked] = useState(false);
   const [presenterIntro, setPresenterIntro] = useState("");
```

Comment:
```suggestion
  const [readMoreText, setReadMoreText] = useState("");
```
this component is genertic

Reply:
fixed